### ISSUE TITLE
Getting control messages for sub-orchestration for non existing instanceId

### ISSUE THREAD

**Issue by ltrain777** [Azure/azure-functions-durable-extension#2010]
### Description
> Checking slowdown issue in production leads me to see control queue messages being created for non-existing instanceId. These subOrchestration instanceId (taskId-guid:actionName) would have been generated by a main orchestrator which instanceId (taskId-guid) doesn't exist either. 

### Expected behavior
> Expecting to see an orchestration instanceId (taskId-guid) in the Instance table which would indicate that Orchestration started

### Actual behavior
> No indication that orchestration started, but somehow the sub-orchestration instanceId was created.

### Relevant source code snippets
```csharp
public async Task<ProjectOutputData> RunOrchestrator(
            [OrchestrationTrigger] IDurableOrchestrationContext context, ILogger log)
        {
            var orchestratorInputs = context.GetInput<OrchestratorInput>();
            var runtimeId = orchestratorInputs.runtimeId;
            var projectInfo = orchestratorInputs.projectInfo;
            // setup...
            using(var cts = new CancellationTokenSource()){
                try{
                    var allActions = projectInfo.project.actions;
                    var runtimeProxy = context.CreateEntityProxy<IRuntime>(runtimeId);
                    var allActionExs = allActions.Select(a => ActionExecution.createActionExecution(runtimeId.EntityKey, a)).ToArray();
                    await runtimeProxy.InitalizeRuntime((projectInfo.userInputs, projectInfo.project.variables, allActionExs));
                    // get first round of actions without dependencies (all rootActions ran in parallel)
                    var nextActions = await context.CallActivityAsync<ActionExecution[]>("getNextActions", Tuple.Create(new List<string>(), new List<string>(), allActionExs, context.InstanceId));
                    var nextActionTasks = new List<Task<RunActionExecutionResult[]>>(); // outside while loop
                    var completedActions = new List<string>();
                    var runningActions = new List<string>();
                    while(true){ 
                        foreach(var nextAction in nextActions)
                        {
                            var actionInstanceId = $"{runtimeId.EntityKey}:{nextAction.action.id}";
                            if(runningActions.IndexOf(nextAction.action.id) == -1){
                                var task = context.CallSubOrchestratorAsync<RunActionExecutionResult[]>("runActionExecution", actionInstanceId, new RunActionExecutionInput(runtimeId, publishBaseUri, expiryTime, maxBatchCount, nextAction, projectInfo.userToken));
                                nextActionTasks.Add(task);
                            }
                        }
                        // Why does this even work? This is why:
                        // https://github.com/Azure/azure-functions-durable-extension/issues/962#issuecomment-539125306
                        Task<RunActionExecutionResult[]> firstFinishedTask = await Task.WhenAny(nextActionTasks);
                        var result = await firstFinishedTask;
                       ...
                        nextActions = await context.CallActivityAsync<ActionExecution[]>("getNextActions", Tuple.Create(completedActions, runningActions, allActionExs, context.InstanceId));
                        if((completedActions.Count /* + skippedActions.Count */)  >= allActions.Length){
                            logWrapper.LogTrace(LoggingConstants.Status.InProgress,
                                                    $"Done running actions: completedActions {completedActions.Count} == allActions {allActions.Length}");
                            cts.Cancel();
                            break;
                        }
                    }
                    // process outputs...
                }catch(){}
                finally{}
            }
        }
```

### Known workarounds
> None 

### App Details
``` xml
<PackageReference Include="Microsoft.Azure.WebJobs.Extensions.CosmosDB" Version="3.0.10" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DurableTask" Version="2.5.0" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.ServiceBus" Version="4.3.0" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="4.0.4" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid" Version="2.1.0" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Host.Storage" Version="4.0.1" />
    <PackageReference Include="Microsoft.Azure.Storage.Blob" Version="11.2.3" />
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.13" />
    <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
    <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.20.1" />
```
- **Azure Functions runtime version **: ~3
- **Programming language used**: C# 

### Screenshots
I've stopped new jobs from coming into this Function app so now it's just trying to process old jobs (where the instance don't exist)
``` console
2021-11-12T23:13:06  Welcome, you are now connected to log-streaming service. The default timeout is 2 hours. Change the timeout with the App Setting SCM_LOGSTREAM_TIMEOUT (in seconds).
2021-11-12T23:14:06  No new trace in the past 1 min(s).
2021-11-12T23:15:06  No new trace in the past 2 min(s).
2021-11-12T23:16:06  No new trace in the past 3 min(s).
2021-11-12T23:17:06  No new trace in the past 4 min(s).
2021-11-12T23:17:27.969 [Warning] 0234c966-861b-4320-90a3-c4e496fe637c:ServerQuery-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-00
2021-11-12T23:17:27.969 [Warning] 0234c966-861b-4320-90a3-c4e496fe637c:ServerQuery-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-00 and setting a visibility delay of 2ms
2021-11-12T23:17:28.172 [Warning] ba57f26d-c78d-4673-94bf-4176a17facc0:cmdbquery-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:28.218 [Warning] 0b6db8e4-4c21-4569-803f-1579ee90d484:companyquery-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-00
2021-11-12T23:17:28.218 [Warning] 0b6db8e4-4c21-4569-803f-1579ee90d484:companyquery-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-00 and setting a visibility delay of 2ms
2021-11-12T23:17:28.264 [Warning] 99cbe910-c3fc-4848-83e9-beb82778440b:ChangeCIState-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:28.310 [Warning] 734edaa0-8806-4e47-8bc6-ca1bf5a11857:ChangeCIState-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:28.356 [Warning] 91342a66-42da-42cd-8c96-1b0319ac0c90:cmdbquery-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:28.402 [Warning] b235bd1d-c495-43b3-9804-8d21be9b8bd9:ChangeCIState-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:28.451 [Warning] 3d2377da-0dcb-4c02-8f95-f39c0bb44785:cmdbquery-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:28.106 [Warning] f23e751f-6813-45a9-9795-ce18c066f5fd:MapIncident-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:28.163 [Warning] 1d463d3d-188e-48ec-b76a-fbfaa38abf55:MapIncident-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-01
2021-11-12T23:17:28.163 [Warning] 1d463d3d-188e-48ec-b76a-fbfaa38abf55:MapIncident-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-01 and setting a visibility delay of 2ms
2021-11-12T23:17:28.213 [Warning] 1b6fc6a1-862d-4f8e-b111-803ac5e1cc56:MapIncident-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-01
2021-11-12T23:17:28.213 [Warning] 1b6fc6a1-862d-4f8e-b111-803ac5e1cc56:MapIncident-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-01 and setting a visibility delay of 2ms
2021-11-12T23:17:28.260 [Warning] b01528f9-a1b7-413f-adbc-01e41f966548:MapIncident-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:28.307 [Warning] adcbfa64-2127-456c-ad98-10b36d1708c7:MapIncident-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:30.442 [Warning] 1b6fc6a1-862d-4f8e-b111-803ac5e1cc56:MapIncident-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-01
2021-11-12T23:17:30.442 [Warning] 1b6fc6a1-862d-4f8e-b111-803ac5e1cc56:MapIncident-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-01 and setting a visibility delay of 4ms
2021-11-12T23:17:30.486 [Warning] 1d463d3d-188e-48ec-b76a-fbfaa38abf55:MapIncident-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-01
2021-11-12T23:17:30.486 [Warning] 1d463d3d-188e-48ec-b76a-fbfaa38abf55:MapIncident-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-01 and setting a visibility delay of 4ms
2021-11-12T23:17:31.646 [Warning] 0234c966-861b-4320-90a3-c4e496fe637c:ServerQuery-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-00
2021-11-12T23:17:31.646 [Warning] 0234c966-861b-4320-90a3-c4e496fe637c:ServerQuery-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-00 and setting a visibility delay of 4ms
2021-11-12T23:17:31.690 [Warning] 0b6db8e4-4c21-4569-803f-1579ee90d484:companyquery-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-00
2021-11-12T23:17:31.690 [Warning] 0b6db8e4-4c21-4569-803f-1579ee90d484:companyquery-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-00 and setting a visibility delay of 4ms
2021-11-12T23:17:37.836 [Warning] 1d463d3d-188e-48ec-b76a-fbfaa38abf55:MapIncident-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-01
2021-11-12T23:17:37.836 [Warning] 1d463d3d-188e-48ec-b76a-fbfaa38abf55:MapIncident-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-01 and setting a visibility delay of 8ms
2021-11-12T23:17:37.883 [Warning] 0234c966-861b-4320-90a3-c4e496fe637c:pstask-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-01
2021-11-12T23:17:37.883 [Warning] 0234c966-861b-4320-90a3-c4e496fe637c:pstask-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-01 and setting a visibility delay of 2ms
2021-11-12T23:17:37.929 [Warning] 1b6fc6a1-862d-4f8e-b111-803ac5e1cc56:MapIncident-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-01
2021-11-12T23:17:37.929 [Warning] 1b6fc6a1-862d-4f8e-b111-803ac5e1cc56:MapIncident-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-01 and setting a visibility delay of 8ms
2021-11-12T23:17:37.975 [Warning] 0b6db8e4-4c21-4569-803f-1579ee90d484:pstask-0: Message [TimerFired#1] was received out of order and will be temporarily returned back to ampautoengineprod3-control-01
2021-11-12T23:17:37.976 [Warning] 0b6db8e4-4c21-4569-803f-1579ee90d484:pstask-0: Abandoning [TimerFired#1] message back to ampautoengineprod3-control-01 and setting a visibility delay of 2ms
2021-11-12T23:17:39.021 [Warning] 734edaa0-8806-4e47-8bc6-ca1bf5a11857:cmdbquery-0: Discarding 1 dequeued history event(s): Instance is Completed
2021-11-12T23:17:39.064 [Warning] b235bd1d-c495-43b3-9804-8d21be9b8bd9:cmdbquery-0: Discarding 1 dequeued history event(s): Instance is Completed
```
![image](https://user-images.githubusercontent.com/1831714/141600944-39abe347-7bef-44bc-b950-f8c92967a985.png)

### If deployed to Azure
> We have access to a lot of telemetry that can help with investigations. Please provide as much of the following information as you can to help us investigate!

- **Timeframe issue observed**: Over time the orchestration takes longer to complete so it's possible that there is something like the a durable entity that is too big. We have a high amount of large-message blobs.
- **Function App name**: https://amp-auto-engine-25d0b887.azurewebsites.net
- **Function name(s)**: Orchestrator, runActionExecution
- **Azure region**: West US
- **Orchestration instance ID(s)**: 2fe43bb6-5637-41a7-9bdf-1a04e1544627:associateAlert-0
- **Azure storage account name**: storageaccountautoea071
- **HubName**: AmpAutoEngineProd3

Might be related to #1999

**Comment by ltrain777** [comment]
After looking for errors in the traces in the "DurableTask" category, I found there were some errors that started on 2021-09-15T03:13:41.3438151Z. I couldn't find what the underlying issue that caused these storage errors.

**Comment by amdeel** [comment]
Hey @ltrain777 it sounds like you experienced a slowdown of processing messages on control queues. Since the version you're using we have implemented a few performance related fixes specifically related to control queue slowdowns. Before I investigate your issue further, can you upgrade to our latest version of Durable Functions v2.6.0 and let us know if this fixes your issue?

**Comment by ltrain777** [comment]
@amdeel I've updated our code to v2.6.0 and deployed to another function app on Friday to get our Production back on track (new DurableTaskHub). I'll update you on whether we see any slowdowns with the new version. I kept the old function app running to complete old request waiting for external events.

Do you see any issues on our old instance that might be causing queue slowdown? What could be causing so many "Abandoning" messages?

**Comment by amdeel** [comment]
@ltrain777 I haven't looked into your specific case closely yet. Upgrading to the latest version of Durable will determine if I need to. We fixed a control queue processing slowdown bug that might be causing your slowdown. As far as sub orchestrators, it's possible your sub orchestrators are completing before your parent completes its first execution and writes itself in history. We also fixed a bug related to this in 2.6.0 which caused SubOrchestratorCompleted messages to be abandoned without being considered out of order messages and being retried. You could be running into both of these problems. After running on 2.6.0 we'll know for sure if you're experiencing a different issue.

**Comment by ltrain777** [comment]
It's been about 9 days running v2.6.0 and I haven't seen any queue issues. I'm going to close the issue.
